source/material_model/rheology/strain_dependent.cc

```diff
@@ -415,7 +489,12 @@ namespace aspect
         if  (this->simulator_is_past_initialization() && this->get_timestep_number() > 0 && in.requests_property(MaterialProperties::reaction_terms))
           {
             const double edot_ii = std::max(sqrt(std::fabs(second_invariant(deviator(in.strain_rate[i])))),min_strain_rate);
-            const double e_ii = edot_ii*this->get_timestep();
+            double e_ii = edot_ii*this->get_timestep();
+
+            // Adjusting strain values to account for strain healing without exceeding an unreasonable range
+            if (healing_mechanism != no_healing)
+              e_ii = std::min(std::max(calculate_strain_healing(in,edot_ii,i),0.0),1.e5);
```

Comment:
Following my comment from above this line should probably be something like:

```suggestion
{
    // Never heal more strain than exists.
     const double healed_strain = std::min(calculate_strain_healing(in,edot_ii,i), e_ii);
      e_ii -= healed_strain;
}
```
From how I understood your strain healing model the returned value should not be negative (i.e. it is always healing, never damaging), so we should not need the upper limiter and only need the lower limit to ensure e_ii stays positive or 0. Is that correct?

Reply:
Yes, I don't want to return a negative value. The upper limiter is there because if the recovery rate is very fast, the strain healing returned becomes physically unreasonable and this stops it.

Reply:
But would that not mean you need to limit this only in one direction (preventing too fast healing)? If so limiting the amount of healing to at most the amount of currently stored strain should be enough.

Reply:
Ah I see, yeah putting e_ii as the other option makes sense.